File: server/src/models/Notification.js
const mongoose = require('mongoose');

const notificationSchema = new mongoose.Schema({
  recipient: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  sender: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  type: {
    type: String,
    enum: ['challenge_received', 'submission_received', 'challenge_approved', 'challenge_rejected', 'friend_request', 'friend_request_accepted', 'truth_received', 'dare_received'],
    required: true
  },
  relatedId: { type: mongoose.Schema.Types.ObjectId }, // Challenge ID, Friend ID, etc.
  message: { type: String, required: true },
  read: { type: Boolean, default: false }
}, { timestamps: true });

notificationSchema.index({ recipient: 1, createdAt: -1 });

module.exports = mongoose.model('Notification', notificationSchema);

File: lib/models/notification.dart
import 'user.dart';

class NotificationModel {
  final String id;
  final String message;
  final String type;
  final bool read;
  final DateTime createdAt;
  final User? sender;

  NotificationModel({
    required this.id,
    required this.message,
    required this.type,
    required this.read,
    required this.createdAt,
    this.sender,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['_id'] ?? '',
      message: json['message'] ?? '',
      type: json['type'] ?? '',
      read: json['read'] ?? false,
      createdAt: DateTime.parse(json['createdAt']),
      sender: json['sender'] != null ? User.fromJson(json['sender']) : null,
    );
  }
}
